app/scripts/controllers/preferences.js

```diff
@@ -118,9 +118,6 @@ export default class PreferencesController {
     this.store.setMaxListeners(13);
     this.tokenListController = opts.tokenListController;
 
-    // subscribe to account removal
-    opts.onAccountRemoved((address) => this.removeAddress(address));
-
```

Comment:
Why is necessary to remove this?

Reply:
The `_onKeyringControllerUpdate` and `onAccountRemoved` both would be fired when an account is removed. This leads to an error in the background and also causing the test to fail. The issue was brought up here, and would be fixed in this pr as well.

https://consensys.slack.com/archives/GTQAGKY5V/p1699007928401829